src/buildApplication/getVercelConfig.ts

```diff
@@ -28,6 +28,57 @@ export async function getVercelConfig(): Promise<VercelConfig> {
 	return config;
 }
 
+/**
+ * Gets the routes of a specific vercel phase
+ *
+ * @param config Vercel config object containing all the routes information
+ * @param phase the phase from which extract the routes
+ * @returns the phases' routes
+ */
+export function getPhaseRoutes(
+	config: VercelConfig,
+	phase: VercelPhase
+): VercelRoute[] {
+	const routes: VercelRoute[] = [];
+
+	if (!config.routes) {
+		return [];
+	}
+
+	let currentRouteIdx = 0;
+
+	if (phase === 'none') {
+		while (
```

Comment:
It feels like these while loops (checking for undefined items in the routes array) would be better coded as for-loops stopping on the index being longer than the array. Or is it possible to actually have undefined values in the array?

Reply:
Here is a suggestion of how this could look (and be more self documenting)...

```ts
export function getPhaseRoutes(
	config: VercelConfig,
	phase: VercelPhase
): VercelRoute[] {
	if (config.routes === undefined) {
		return [];
	}
	const phaseStart = findStartOfPhase(config.routes, phase);
	return getRoutesInPhase(config.routes, phaseStart);
}

function findStartOfPhase(routes: VercelRoute[], phase: string) {
	if (phase === 'none') {
		return 0;
	} else {
		const index = routes.findIndex(
			route => isVercelHandler(route) && route.handle === phase
		);
		return index === -1 ? routes.length : index + 1;
	}
}

function getRoutesInPhase(routes: VercelRoute[], phaseStart: number) {
	const collectedRoutes: VercelRoute[] = [];
	for (let i = phaseStart; i < routes.length; i++) {
		const route = routes[i];
		if (!route || isVercelHandler(route)) {
			break;
		}
		collectedRoutes.push(route);
	}
	return collectedRoutes;
}

function isVercelHandler(route: VercelRoute): route is VercelHandler {
	return 'handler' in route;
}
```

Reply:
Thanks! 😄 

although I am really not sure about the two functions `getPhaseRoutes` and `getRoutesInPhase` they basically convey the same exact intent don't they?

what I mean is, if someone unfamiliar with this code would read the two functions names it would be very unclear what each does and how they are different

I've applied your suggestion but merged the two functions into one, I hope that's fine, please have a look 🙏

Reply:
I've also tried improving things further and avoid iterative loops entirely (just using `findIndex`es + `slice` at the end)

please let me know what you think 😄

Reply:
Great! Made one more suggestion.